src/test/java/computerdatabase/AdvancedSimulation.java

```diff
@@ -0,0 +1,95 @@
+import java.util.concurrent.ThreadLocalRandom;
+
+/**
+ * See the Gatling advanced tutorial on <a href="https://gatling.io/docs/gatling/tutorials/advanced/">the Gatling website</a>.
+ */
+public class AdvancedSimulation extends Simulation {
+
+    FeederBuilder<String> feeder = csv("search.csv").random();
+
+    ChainBuilder search =
+        exec(http("Home").get("/"))
+            .pause(1)
+            .feed(feeder)
+            .exec(
+                http("Search")
+                    .get("/computers?f=#{searchCriterion}")
+                    .check(
+                        css("a:contains('#{searchComputerName}')", "href").saveAs("computerUrl")
+                    )
+            )
+            .pause(1)
+            .exec(http("Select").get("#{computerUrl}").check(status().is(200)))
+            .pause(1);
+
+    // repeat is a loop resolved at RUNTIME
+    ChainBuilder browse =
+        // Note how we force the counter name so we can reuse it
+        repeat(4, "i").on(
+            exec(
+                http("Page #{i}")
+                    .get("/computers?p=#{i}")
+            ).pause(1)
+        );
+
+    // Note we should be using a feeder here
+    // let's demonstrate how we can retry: let's make the request fail randomly and retry a given
+    // number of times
+
+    ChainBuilder edit =
+        // let's try at max 2 times
+        tryMax(2)
+            .on(
+                exec(http("Form").get("/computers/new"))
+                    .pause(1)
+                    .exec(
+                        http("Post")
+                            .post("/computers")
+                            .formParam("name", "Beautiful Computer")
+                            .formParam("introduced", "2012-05-30")
+                            .formParam("discontinued", "")
+                            .formParam("company", "37")
+                            .check(
+                                status().is(
+                                    // we do a check on a condition that's been customized with
+                                    // a lambda. It will be evaluated every time a user executes
+                                    // the request
+                                    session -> 200 + ThreadLocalRandom.current().nextInt(2)
+                                )
+                            )
+                    )
+            )
+            // if the chain didn't finally succeed, have the user exit the whole scenario
+            .exitHereIfFailed();
+
+    HttpProtocolBuilder httpProtocol =
+        http.baseUrl("http://computer-database.gatling.io")
+            .acceptHeader("text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8")
+            .doNotTrackHeader("1")
+            .acceptLanguageHeader("en-US,en;q=0.5")
+            .acceptEncodingHeader("gzip, deflate")
+            .userAgentHeader(
+                "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.8; rv:16.0) Gecko/20100101 Firefox/16.0"
```

Comment:
I think we should keep the simulation as small as possible, and remove the "useless" header: userAgent, acceptLanguage, doNotTrack

Reply:
We can remove the stupid and useless DNT.
However, we definitely want to keep the other ones.
Remember that those change the server behavior. We want to educate the users about them.